### PR TITLE
refactor: Migrate AssetService to async/await

### PR DESCRIPTION
**Pull request by bennyn**

### CODE REVIEW

File: src/script/util/util.ts [hunk 1]
```diff
@@ -135,12 +135,12 @@ export const loadUrlBuffer = (
...
 };
 
-export const loadImage = function (blob: Blob): Promise<GlobalEventHandlers> {
+export const loadImage = function (blob: Blob): Promise<HTMLImageElement> {
   return new Promise((resolve, reject) => {
     const object_url = window.URL.createObjectURL(blob);
     const img = new Image();
     img.onload = function (): void {
-      resolve(this);
+      resolve(img);
```
Comment (ffflorian) [inline src/script/util/util.ts:143]: If this works, you can use an arrow function in the line above.

File: src/script/assets/AssetService.ts [hunk 2]
```diff
@@ -231,23 +221,21 @@ export class AssetService {
...
-      })
-      .then(compressedBytes => {
-        return loadImage(new Blob([compressedBytes], {type: image.type})).then(compressedImage => ({
-          compressedBytes,
-          compressedImage,
-        }));
-      });
+    const buffer = await loadFileBuffer(image);
+    if (typeof filter === 'function' ? filter() : undefined) {
+      new Uint8Array(buffer as ArrayBuffer);
```
Comment (ffflorian) [inline src/script/assets/AssetService.ts]: There is something missing?

Comment (bennycode) [reply src/script/assets/AssetService.ts]: Good catch, I reverted it.

File: src/script/assets/AssetService.ts [hunk 3]
```diff
@@ -231,23 +221,23 @@ export class AssetService {
...
-      })
-      .then(compressedBytes => {
-        return loadImage(new Blob([compressedBytes], {type: image.type})).then(compressedImage => ({
-          compressedBytes,
-          compressedImage,
-        }));
-      });
+    const buffer = await loadFileBuffer(image);
+    let compressedBytes: ArrayBuffer;
+    if (typeof filter === 'function' ? filter() : undefined) {
```
Comment (ffflorian) [inline src/script/assets/AssetService.ts]: IMO it makes more sense to make clear that, if no `filter` is available, this should not run:

```suggestion
    if (typeof filter === 'function' ? filter() : false) {
```

What do you think?

**Pull request merged**